src/torchmetrics/functional/retrieval/average_precision.py

```diff
@@ -11,13 +11,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Optional
+
 import torch
 from torch import Tensor, tensor
 
 from torchmetrics.utilities.checks import _check_retrieval_functional_inputs
 
 
-def retrieval_average_precision(preds: Tensor, target: Tensor) -> Tensor:
+def retrieval_average_precision(preds: Tensor, target: Tensor, top_k: Optional[int] = None) -> Tensor:
```

Comment:
Consider the fact that other `RetrievalMetric`s use `k` instead of `top_k`. 
Got the sake of consistency it might be better to stick with `k` here?

Reply:
let's stick with `top_k`. The `RetrievalMetric` is updated in #1504

Reply:
Ah, I wasn't aware of that PR. Thanks @justusschock